examples/demo/vite.config.ts

```diff
@@ -68,6 +68,21 @@ export default defineConfig(async () => {
                 //     find: 'scheduler/tracing',
                 //     replacement: 'scheduler/tracing-profiling',
                 // },
+                // The 2 next aliases are needed to avoid having multiple MUI instances
+                {
+                    find: '@mui/material',
+                    replacement: path.resolve(
+                        __dirname,
+                        'node_modules/@mui/material'
+                    ),
+                },
+                {
+                    find: '@mui/icons-material',
+                    replacement: path.resolve(
+                        __dirname,
+                        'node_modules/@mui/icons-material'
+                    ),
+                },
                 // we need to manually follow the symlinks for local packages to allow deep HMR
```

Comment:
How about the core MUI packages, which are indirect dependencies ? And how about other major dependencies overridden in the demo `package.json` (react and react-router)?

Reply:
The core MUI packages are resolved from the MUI packages that require them. In our case, the packages from the demo node_modules. I haven't seen multiple react-router packages loaded (nor in vite, nor in the node_modules)